[P1-N10] Warn users against re-using salts (#1273)
UMAprotocol/protocol: documentation/oracle/dvm_interfaces.md

`Commit`.

It also takes the `hash` that the voter wishes to commit to. The `hash` is created by computing

> `keccak256(price, salt, time, address, roundId, identifier)`, where price is the `int256` price value that they wish to submit, `time` is the unix timestamp of the request being voted on, `address` is the voter's address (technically, the same one that will reveal the vote), `roundId` is the current voting round, `identifier` is the pricefeed identifier relevant to the price request, and `salt` is a random `int256` value. The voter must remember the inputs to the vote `hash` that they submitted so they can reveal their commit later - otherwise, the commit cannot be revealed and the vote won't be counted.

Author: @rcai1 I changed this description to `commitVote` to reflect that  the vote hash now requires additional information.

Reviewer: Makes sense!

Since transaction data is public, the salt will be revealed along with the vote. While this is the system’s expected behavior,

> voters should never reuse salts. If someone else is able to predict the voted price or knows that a price will be reused, then

Reviewer 2: ```suggestion
voters should never reuse salts. If someone else is able to guess the voted price and knows that a salt will be reused, then
```

> they can easily reveal the vote.

Reviewer 2: ```suggestion
they can easily determine the vote pre-reveal.
```

This method takes the `identifier` and `time` to identify the price request.

So the reveal can be verified on-chain, the voter must also provide the `price` and `salt` that they used to compute

> the `hash` that they passed to `commitVote`. Since transaction data is public, the salt will be revealed with the vote. While this is the system’s expected behavior,

Reviewer 2: nit: maybe we should also put this warning in the commitVote section.

Author: good idea done.

> it assumes that voters will never reuse the same pair of salt and price. Voters should never reuse pairs of salt and price or their commits can be disclosed in advance, and to be safe voters should avoid reusing salts in general.

Reviewer 2: Same comment as above.
